### PR TITLE
Flow vs TypeScript

### PR DESCRIPTION
**Pull request by aghh1504**

### CODE REVIEW

File: proposals/flow-vs-typescript.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+2. Typescript provides to you some additional language services such as code completion features, navigation and refactoring features whereas flow aims to build a deeper level of understanding of your code and is responsible for doing an interprocedural analysis.
+
+3. Difference with flow and TS syntax here - https://github.com/niieani/typescript-vs-flowtype
+
+4. TypeScript wants to provide great tooling and language services for autocompletion, code navigation, and refactoring. Flow, on the other hand, develops a deeper understanding of your code and even does interprocedural analyses.
+
+5. TypeScript implements both a type checker and a transpiler that emits plain JavaScript. Flow only does type checking and relies on Babel or flow-remove-types or some other tool to remove type annotations.
+
+# How easy would it be to migrate from flow to typescript?
```
[the remark this answers is not in the document]
Comment (aghh1504) [reply proposals/flow-vs-typescript.md]: o thats sounds cool

**Review by cutandpastey — DISMISSED** [review]

File: proposals/flow-vs-typescript.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+## Cons:
+
+1. Lesser React support because its great for Angular project. But nowdays its getting better and better.
+
+2. Less flexible then Flow.
+
+#Key Differences Between TypeScript vs flow
+
+1. Flow is any day a better choice to go when you have to work with type checking static kind functionalities without even writing the non-standard Javascript code i.e. the code which asks for compilation back into Javascript. To use this feature, you can write type annotations in comments rather than using them in the executable code itself.
```
Comment (dbbk) [inline proposals/flow-vs-typescript.md]: I don't really understand what this means, can it be explained any clearer?

File: proposals/flow-vs-typescript.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+Flow and TypeScript are both fantastic ways to limit technical debt in your JavaScript projects, allowing you to, among other things, evolve your codebase with confidence.
+They are similar tools that focus on static types checking.
+
+# Flow
+
+## Pros:
+
+1. As you're using React, we might prefer Flow because it easily integrates with Babel and other infrastructure.
+
+2. Flow is known to be a simpler language than TypeScript due to it’s out of the box utility. A developer would have to learn TypeScript syntax e.g. Type Annotation and how to modify the code.
```
Comment (dbbk) [inline proposals/flow-vs-typescript.md]: Isn't this an equal point for both Flow and TypeScript? They both have their own type annotation syntax

File: proposals/flow-vs-typescript.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+4. More solid type checking and less changes when upgrade to newest version.
+
+## Cons:
+
+1. Lesser React support because its great for Angular project. But nowdays its getting better and better.
+
+2. Less flexible then Flow.
+
+#Key Differences Between TypeScript vs flow
```
Comment (dbbk) [inline proposals/flow-vs-typescript.md]: This heading is broken, needs a space after the hash

**Review by benkingcode — CHANGES_REQUESTED** [review]

File: proposals/flow-vs-typescript.md (hunk 4 above)
Comment (aghh1504) [reply proposals/flow-vs-typescript.md]: ok

**Review by benkingcode — DISMISSED** [review]

File: proposals/flow-vs-typescript.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+2. Not all utility types from Flow are documented. This might be fixed soon though.
+
+# TypeScript
+
+## Pros:
+
+1. Larger community: Because it is an older project its community is larger.
+
+2. Faster then flow.
```
Comment (emuraton) [inline proposals/flow-vs-typescript.md]: typo `than*`

File: proposals/flow-vs-typescript.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+3. Upfront compile errors: This can probably be configured in both TypeScript and Flow, but having upfront compilation errors instead of a non mandatory type check seems to save time when testing your code by preventing you from running code that is bogus anyways.
+
+4. More solid type checking and less changes when upgrade to newest version.
+
+## Cons:
+
+1. Lesser React support because its great for Angular project. But nowdays its getting better and better.
+
+2. Less flexible then Flow.
```
Comment (emuraton) [inline proposals/flow-vs-typescript.md]: also

**Review by emuraton — DISMISSED** [review]

**Review by cutandpastey — APPROVED** [review]

**Pull request merged**